plugins/artifacts/payload-schema.yml

```diff
@@ -0,0 +1,30 @@
+$schema: http://json-schema.org/draft-04/schema#
+title: payload
+description: Artifacts to be published
+type: array
+items:
+  type: object
+  properties:
+    type:
+      title: Upload type
+      description: Artifacts can be either an individual `file` or a `directory` containing potentially multiple files with recursively included subdirectories
+      type: string
+      enum: [file, directory]
+    path:
+      title: Artifact Path
+      description: Filesystem path of artifact
+      type: string
+      pattern: "^.*[^/]$"
+    name:
+      title: Artifact Name
+      description: This will be the leading path to directories and the full name for files that are uploaded to s3
+      type: string
+      pattern: "^[^/].*[^/]$"
```

Comment:
@djmitche, @gregarndt If we ever want to enforce artifact naming... Restrict characters to printable ascii or something like that, now is the time...
We can't really make this change on `queue.taskcluster.net` before we've made sure that all uploaders don't violate this requirement...

However, given that the scope `'queue:get-artifact:<name>'` is required to get an artifact... We are already in a situation where any non-public artifact can't be downloaded if name contains characters that isn't ascii printable.

This might be the wrong the place for this discussion, but perhaps we should take the first step in disallowing non-printable ascii chars in artifact names...
I doubt any are in use, but we should probably do a table scan and check before we change the queue. However, locking it down here is a good start.